app.rb

```diff
@@ -16,15 +59,280 @@ class MyApp < Sinatra::Base
     end
   end
 
-  error do |e|
-    if e.is_a? ActiveRecord::RecordNotFound
-      halt 404, json(error: "Not Found")
-    elsif e.is_a? ActiveRecord::RecordInvalid
-      halt 422, json(error: e.message)
+
+  def require_login!
+    unless current_user
+    redirect '/'
+    end
+  end
+
+  before do
+    unless ["/", "/newuser", "/dashboard"].include?(request.path)
```

Comment:
Question: how do we feel about having e.g. a `PUBLIC_ROUTES` variable that we check here? Then we could write tests that say something like: when we visit any of the public routes (and aren't logged in), nothing errors.